test/test.jl

```diff
@@ -13,6 +13,19 @@
         @test_log(logger, "info", x -> x == msg, Memento.info(logger, msg))
     end
 
+    @testset "@test_nolog" begin
+        logger = getlogger("test_nolog")
+        setlevel!(logger, "info")
+
+        msg = "Hello!"
+        different_msg = "Goodbye"
+
+        @test_nolog(logger, "info", msg, nothing)
+        @test_nolog(logger, "info", msg, Memento.debug(logger, msg))
+        @test_nolog(logger, "info", msg, Memento.info(logger, different_msg))
+        @test_nolog(logger, "info", msg, Memento.warn(logger, different_msg))
```

Comment:
Might be good to include a test to check expected failure cases with an `@test_throws`?

Reply:
I thought the same but couldn't think of a way to do that. `@test_throws` doesn't catch the test failure (see stacktrace below), it might if we used `@assert` but that defeats the purpose of `@test_nolog` being a testing macro?

```julia
@test_nolog: Test Failed at /Users/nicoleepp/.julia/dev/Memento/src/memento_test.jl:64
  Expression: handler.found[1] == EMPTY_MATCH && handler.found[2] == EMPTY_MATCH
Stacktrace:
 [1] (::getfield(Main, Symbol("##38#53")))() at /Users/nicoleepp/.julia/dev/Memento/src/memento_test.jl:64
 [2] setpropagating!(::getfield(Main, Symbol("##38#53")), ::Logger, ::Bool) at /Users/nicoleepp/.julia/dev/Memento/src/loggers.jl:113
 [3] macro expansion at /Users/nicoleepp/.julia/dev/Memento/src/memento_test.jl:62 [inlined]
 [4] macro expansion at /Users/nicoleepp/.julia/dev/Memento/test/test.jl:23 [inlined]
 [5] macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
 [6] macro expansion at /Users/nicoleepp/.julia/dev/Memento/test/test.jl:17 [inlined]
 [7] macro expansion at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Test/src/Test.jl:1083 [inlined]
 [8] top-level scope at /Users/nicoleepp/.julia/dev/Memento/test/test.jl:2
@test_nolog: Test Failed at /Users/nicoleepp/.julia/dev/Memento/test/test.jl:23
  Expression: #= /Users/nicoleepp/.julia/dev/Memento/test/test.jl:23 =# @test_nolog logger "info" msg Memento.info(logger, msg)
    Expected: ErrorException
  No exception thrown
```